### ISSUE TITLE
Different behaviors when using  relu activation inside conv2d layer and a standalone ReLU() layer

### ISSUE THREAD

**Issue by srihari-humbarwadi** [tensorflow/tensorflow#22462]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: No
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.10.1
- **Python version**: 3,6
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: 9.0/7.0
- **GPU model and memory**: GTX 1080ti 11gb
- **Exact command to reproduce**: run the code on mnist dataset

```
from tensorflow.python.keras.layers import Conv2D, ReLU, BatchNormalization, Dense, Input, Conv2DTranspose, UpSampling2D, Flatten, Reshape
from tensorflow.python.keras.models import Model
from tensorflow.python.keras.utils import multi_gpu_model
from tensorflow.python.keras.optimizers import Nadam
from tensorflow.python.keras.callbacks import TensorBoard, ModelCheckpoint, EarlyStopping, ReduceLROnPlateau
import tensorflow as tf
import numpy as np
from tensorflow.python.keras import backend as K
from tensorflow.python.keras.datasets import mnist


img_height, img_width = 28, 28
(x_train, y_train), (x_test, y_test) = mnist.load_data()
X = np.expand_dims(x_train, axis=-1)
print(X.shape)

input_image = Input(shape=(img_height, img_width, 1), name='image_imput')
x = Conv2D(filters=32, kernel_size=(5, 5), strides=(2, 2), padding='same', name ='encoder_conv1', activation='relu')(input_image)
x = Conv2D(filters=64, kernel_size=(5, 5), strides=(2, 2), padding='same', name ='encoder_conv2', activation='relu')(x)
x = Conv2D(filters=128, kernel_size=(3, 3), strides=(2, 2), padding='valid', name ='encoder_conv3', activation='relu')(x)
x = Flatten()(x)
encoded = Dense(units=10)(x)

y = Dense(units=1152, activation='relu')(encoded)
y = Reshape((3, 3, 128))(y)
y = Conv2DTranspose(filters=64, kernel_size=(3, 3), strides=(2, 2), padding='valid', name ='decoder_deconv1', activation='relu')(y)
y = Conv2DTranspose(filters=32, kernel_size=(5, 5), strides=(2, 2), padding='same', name ='decoder_deconv2', activation='relu')(y)
decoded_image = Conv2DTranspose(filters=1, kernel_size=(5, 5), strides=(2, 2), padding='same', name ='decoder_deconv3', activation='relu')(y)


CAE = Model(inputs = input_image, outputs = decoded_image, name = 'CAE')


tb = TensorBoard(log_dir='logs', write_graph=True)
mc = ModelCheckpoint(filepath='models/top_weights.h5', monitor='acc', save_best_only='True', save_weights_only='True', verbose=1)
es = EarlyStopping(monitor='loss', patience=15, verbose=1)
rlr = ReduceLROnPlateau(monitor='loss')
callbacks = [tb, mc, es, rlr]
CAE.compile(optimizer='adam', loss='mse', metrics=['accuracy'])


# CAE.load_weights('models/top_weights.h5')
# CAE.save('CAE.h5')
CAE.fit(X, X, epochs=1000, batch_size=256, callbacks=callbacks)
 ```

in the above code if i do something like this that is  use a standalone activation layer, the model behaves differently 
```
x = Conv2D(filters=32, kernel_size=(5, 5), strides=(2, 2), padding='same', name ='encoder_conv1')(input_image)
x = ReLU()(x)
```
If i include the activation in the conv2d layer, the model converges at 80% accuracy and when i use a standalone activation layer the model is stuck at 11% accuracy.

I want to know the reason for different behavior

**Comment by ymodak** [comment]
@srihari-humbarwadi If you don't specify activation argument in the Conv2D layer it will use linear activation by default. Are you specifying separate activation after each convolution layer or just after the first one?

**Comment by srihari-humbarwadi** [comment]
I tried adding a separate activation layer after each conv2d layer. And isn't the activation=None by default?

**Comment by ymodak** [comment]
Okay. Yes it is none by default but according to [keras documentation](https://keras.io/layers/convolutional/) linear activation is applied in the case of absence of activation argument.
> activation: Activation function to use (see activations). If you don't specify anything, no activation is >applied ie. "linear" activation:

**Comment by srihari-humbarwadi** [comment]
Yea but a(x) = x  is just an identity mapping shouldn't it do nothing in this case?
```
activation: Activation function to use (see activations). If you don't specify anything, no activation is applied (ie. "linear" activation: a(x) = x).
```

**Comment by ymodak** [comment]
To confirm this can you try running your model like this : 
x = Conv2D(filters=32, kernel_size=(5, 5), strides=(2, 2), padding='same', name ='encoder_conv1')(input_image)  # run with no activation layer
# x = ReLU()(x)   (comment out the ReLU layer)
After this can you compare the accuracy and share your findings?

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
